delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/stream/pipes/SelectPredicates.scala

```diff
@@ -27,17 +27,21 @@ class SelectPredicates(config: SelectPredicatesConfig) extends Pipe {
   override def inType: Typeable[GraphResource]  = Typeable[GraphResource]
   override def outType: Typeable[GraphResource] = Typeable[GraphResource]
 
-  override def apply(element: SuccessElem[GraphResource]): Task[Elem[GraphResource]] = {
-    val id       = subject(element.value.id)
-    val newGraph = element.value.graph.filter { case (s, p, _) => s == id && config.nodeSet.contains(p) }
-    val newState = element.value.copy(graph = newGraph, types = newGraph.rootTypes)
-    Task.pure(element.copy(value = newState))
+  override def apply(element: SuccessElem[GraphResource]): Task[Elem[GraphResource]] = Task.pure {
+    if (config.forwardTypes.exists { p => p.exists(element.value.types.contains) }) {
```

Comment:
Maybe a name for this predicate like `isAForwardType` is useful here?

Reply:
Invariant type sounds nice

Reply:
I will let as it is for now